corporate/views/remote_billing_page.py

```diff
@@ -164,47 +181,71 @@ def remote_realm_billing_finalize_login(
             remote_user.tos_version.split(".")[0]
         )
     except RemoteRealmBillingUser.DoesNotExist:
-        # This is the first time this user is logging in, so ToS consent needed.
+        # This is the first time this user is logging in.
+        remote_user = None
         tos_consent_needed = True
 
     if request.method == "GET":
-        context = {
-            "remote_server_uuid": remote_server_uuid,
-            "remote_realm_uuid": remote_realm_uuid,
-            "host": remote_realm.host,
-            "user_email": user_email,
-            "user_full_name": user_full_name,
-            "tos_consent_needed": tos_consent_needed,
-            "action_url": reverse(
-                remote_realm_billing_finalize_login, args=(signed_billing_access_token,)
-            ),
-        }
-        return render(
-            request,
-            "corporate/remote_realm_billing_finalize_login_confirmation.html",
-            context=context,
-        )
+        if remote_user is not None:
+            # Render a template where the user will just confirm their info,
+            # possibly accept ToS if needed, POST back here and will get fully
+            # authenticated.
+            context = {
+                "remote_server_uuid": remote_server_uuid,
+                "remote_realm_uuid": remote_realm_uuid,
+                "host": remote_realm.host,
+                "user_email": remote_user.email,
+                "user_full_name": remote_user.full_name,
+                "tos_consent_needed": tos_consent_needed,
+                "action_url": reverse(
+                    remote_realm_billing_finalize_login, args=(signed_billing_access_token,)
+                ),
+            }
+            return render(
+                request,
+                "corporate/remote_realm_billing_finalize_login_confirmation.html",
+                context=context,
+            )
+        else:
+            # This user is logging in for the first time, so we need to create their
+            # RemoteRealmBillingUser object. Render a template where they'll
+            # enter their email address - we'll send a verification email to it.
+            context = {
+                "email": user_email,
+                "action_url": reverse(
+                    remote_realm_billing_confirm_email, args=(signed_billing_access_token,)
+                ),
+            }
+            return render(
+                request,
+                # TODO: We're re-using a template originally made for the legacy server
+                # flow. We should rename it and its HTML contents to a more general name.
+                "corporate/remote_billing_legacy_server_confirm_login_form.html",
```

Comment:
These TODOs are fixed in the 2nd commit. Didn't want to confuse this commit with even more template chaos